tdp/core/deployment/deployment_iterator.py

```diff
@@ -106,115 +108,31 @@ def __init__(
         except NothingToReconfigureError:
             self._reconfigure_operations = None
 
-    def __next__(self) -> Optional[list[SCHStatusLogModel]]:
+    def __next__(
+        self,
+    ) -> Callable[[], Optional[list[SCHStatusLogModel]]]:
         try:
             while True:
-                operation_rec: OperationModel = next(self._iter)
-
-                # Return early if deployment failed
-                if self.deployment.state == DeploymentStateEnum.FAILURE:
-                    operation_rec.state = OperationStateEnum.HELD
-                    return
-                else:
-                    operation_rec.state = OperationStateEnum.RUNNING
+                operation_rec = next(self._iter)
 
                 # Retrieve operation to access parsed attributes and playbook
                 operation = self._collections.get_operation(operation_rec.operation)
 
-                # Run the operation
-                if operation.noop:
-                    # A noop operation is always successful
-                    operation_rec.state = OperationStateEnum.SUCCESS
-                else:
-                    self._run_operation(operation_rec)
-
-                # Set deployment status to failure if the operation failed
-                if operation_rec.state != OperationStateEnum.SUCCESS:
-                    self.deployment.end_time = datetime.utcnow()
-                    self.deployment.state = DeploymentStateEnum.FAILURE
-                    # Return early as status is not updated
-                    return
-
-                # ===== Update the cluster status if success =====
-
-                # Skip sleep operation
-                if operation.name == OPERATION_SLEEP_NAME:
-                    return
-
-                sch_status_logs: list[SCHStatusLogModel] = []
-                sc_name = ServiceComponentName(
-                    service_name=operation.service_name,
-                    component_name=operation.component_name,
+                operation_rec.state = OperationStateEnum.RUNNING
+                if not operation.noop:
+                    operation_rec.start_time = datetime.utcnow()
```

Comment:
This is not ok, if the operation is `HELD` or `FAILURE`, it should not have a `start_date`.

In that case, you shouldn't generate the `-process_operation_fn` either.